src/Facebook/Url/FacebookUrlDetectionHandler.php

```diff
@@ -160,4 +160,24 @@ protected function getHeader($key)
     {
         return $this->getServerVar('HTTP_' . $key);
     }
+
+    /**
+     * Checks if the value in X_FORWARDED_HOST is a valid hostname
+     * Could prevent unintended redirections
+     */
+    protected function isValidForwardedHost()
+    {
+        $host = $this->getHeader('X_FORWARDED_HOST');
+        if (!$host) {
+          return false;
+        }
+
+        $elements = explode(',', $host);
+        $host = $elements[count($elements) - 1];
+        
+        return return (preg_match("/^([a-z\d](-*[a-z\d])*)(\.([a-z\d](-*[a-z\d])*))*$/i", $domain_name) //valid chars check
```

Comment:
one `return` must be removed. BTW, parenthesis arround all conditions can be removed